diff --git a/repair.py b/repair.py
index b37ecef..9cc0652 100644
--- a/repair.py
+++ b/repair.py
@@ -129,6 +129,12 @@ def copylib(src_path, dest_dir, patcher):
 
     if any(itertools.chain(rpaths['rpaths'], rpaths['runpaths'])):
         patcher.set_rpath(dest_path, dest_dir)
+   
+    old_rpaths = patcher.get_rpath(dest_path).split(':') 
+    if '$ORIGIN' not in old_rpaths:
+        rpath_set = OrderedDict([(old_rpath, '') for old_rpath in old_rpaths])
+        rpath_set['$ORIGIN'] = ''
+        patcher.set_rpath(dest_path, ':'.join(rpath_set).strip(':'))
 
     return new_soname, dest_path
 
